61.md

```diff
@@ -0,0 +1,193 @@
+NIP-61
+======
+
+Event Set
+---------
+
+`draft` `optional`
+
+An event set is represented by a group of events, not a single event.
+Each event of a specific set has the same pubkey and `n` (set "n"ame) tag.
+
+An event can have many `n` tags, thus being member of many event sets.
+To remove an event from a set, remove the corresponding `n` tag from the event or
+send a deletion event (`kind:5`).
+
+## Referencing
+
+An event set is referenced by an `s` tag. It
+references many events from an author instead of a single one:
+
+`["s", "<kind>:<pubkey>:<n-tag>", "<recommended-relay-URL, optional>"]`
+
+One may also use an `nset` [NIP-19](19.md) entity through a [NIP-21](21.md) URI: `nostr:nset1qqstn...794234d`.
+
+## How to add items to a set
+
+Just add an `n` tag to your event. For example, adding one of your long-form posts to a set named "a-set-name-example":
+
+```js
+{
+  "kind": 30023,
+  "tags": [
+    ["d", "<random>"],
+    ["n", "a-set-name-example"]
+    // other tags
+  ]
+  // other fields
+}
+```
+
+## Tagging into Event Sets
+
+You SHOULD use a list of pre-defined event kinds to tag existing events, public keys, or other references into sets.
+
+The kind and `d` tag value will vary depending on the type of item being referenced.
+
+**Some** of the event kinds **require** a `k` tag set to the referenced item kind to help with filtering or other tags.
+
+**Table of "Set Item Reference" Event Kinds and Structure by Referenced Item Type**:
+
+| Type | Kind | d-tag | other required tags |
+|-|-|-|-|
+| **pubkey** | 30382 | `<user-pubkey>` | |
+| **event** | 30383 | `<event-pubkey>:<event-id>` | `["k", "<event-kind>"]` |
```

Comment:
```suggestion
| **event** | 30383 | `<event-id>` | `["k", "<event-kind>"]` |
```

Including the pubkey is a painful annoyance. It makes it harder to query, and harder to create the event, with no apparent benefits.

Reply:
I saw in another thread, the justification was that it can be used like a relay hint. That's not a good enough reason to put it in the d-tag value. You could use `["d", <event-id>, <optional relay>]` for the same purpose, or even `["d", <event-id>], ["relay", <relay>]`

Reply:
Let's consider it a fact that both the referenced event author's pubkey and the regular relay hint are important when searching for the referenced event. The former is slower cause needs more steps to find the event, while the latter may suffer from link rot.

With that said, this was my jerk move to force people to add NIP-65 pubkey hint. I agree with the "annoyance" part if you meant that this shouldn't be placed in the `d-tag` cause it shouldn't be indexed by relays.

Besides the `["relay", "<relay>"]` what do you think of `["nip65pk", "<pubkey">]` for the pubkey hint?

Reply:
In the spirit of the `k` tag, I ended up adding a `p` tag of the original author in my events. Here's an example:

```json
{
  "kind": 30383,
  "id": "00dab96e90bf45215b6c8b7e2d6c238977d5395eeeaf7b236d161875537f819f",
  "pubkey": "d4ac0aa9df4ae9c491fde58ae4846079ad20332a59d7a360a186122320ad780f",
  "created_at": 1717945869,
  "tags": [
    ["d", "b8d54511a88165d8e59d29617d66d76c4564e013bdcfec021d663bf63d3b46fe"],
    ["n", "closed"],
    ["p", "0461fcbecc4c3374439932d6b8f11269ccdb7cc973ad7a50ae362db135a474dd"],
    ["k", "1984"],
    ["P", "9989500413fb756d8437912cc32be0730dbe1bfc6b5d2eef759e1456c239f905"]
  ],
  "content": "",
  "sig": "14bc909d5ea13658833d0996ad09185ac5ffbf0bd2ab7f5a9f1a2f02df79e6bda212916258de1039c0ff695b28366f77e9b9ebd0811b26e04dbb1c7f46e58227"
}
```

Reply:
It works but the `k` tag is indexable cause there is a need to get just `30383` events referencing events of specific kinds.
I couldn't think of an use case where one wants to filter also by referenced event pubkey. We shouldn't add one-letter tags for no reason.

And what's the uppercase `P` for?

Reply:
Every time a report comes in, I generate a kind 30383 event for it with `["n", "open"]`. Then instead of querying kind 1984 events, I query `{ kinds: [30383], authors: [me], '#k': [1984] }` and paginate through this list.

This allows me to mark reports as open or closed, and then query by the n tag to get reports depending on their status.

I want the "p" tag so I can also filter by the user who created the report, in conjunction with filters like "open"/"closed".

The uppercase "P" is people who the report tags, so I can similarly filter reports targeting specific users users along with "open"/"closed".

Reply:
> I want the "p" tag so I can...

Thx for the example, changed 30383 and 31383 to use just the event id in the d-tag. Added the `p` tag too.